src/libponyc/pass/names.c

```diff
@@ -199,6 +199,8 @@ static bool names_typeparam(pass_opt_t* opt, ast_t** astp, ast_t* def)
       TREE(cap)
       TREE(ephemeral)));
 
+  ast = *astp;
+
```

Comment:
is there a known bug from this?

Reply:
not that i'm aware of.. just something that the memory sanitizer flagged (iirc)..

Reply:
https://github.com/ponylang/ponyc/pull/4360 created for this